### PR TITLE
Pythagorean tree: Add tests for widget

### PR DESCRIPTION
**Pull request by pavlin-policar**
Add widget tests for Pythagorean tree widget and fixup some deprecation warnings from `gui` library.

### CODE REVIEW

File: Orange/widgets/visualize/tests/test_owpythagorastree.py [hunk 1]
```diff
@@ -73,3 +83,90 @@ def test_compute_center_with_complex_square_angle_with_base_angle(self):
...
+
+
+class TestOWPythagorasTree(WidgetTest):
+    def setUp(self):
+        self.widget = self.create_widget(OWPythagorasTree)
+
+        # Any widget extending `BaseLearner` automatically adds the instances
+        # to the classifier
+        titanic_data = Table('titanic')
+        self.titanic = TreeLearner(max_depth=2)(titanic_data)
```
Comment (kernc) [inline Orange/widgets/visualize/tests/test_owpythagorastree.py]: Create widget in `setUp()`,  build models in `setUpClass()`?

File: Orange/widgets/visualize/tests/test_owpythagorastree.py [hunk 2]
```diff
@@ -73,3 +83,90 @@ def test_compute_center_with_complex_square_angle_with_base_angle(self):
...
+    def test_input_regression_tree(self):
+        # Tree model must be none before assignment
+        self.assertIsNone(self.widget.model)
+        self.assertIsNone(self.widget.tree_adapter)
+
+        # Test classification tree assingment
+        self.send_signal('Tree', self.housing)
+        self.assertIsNotNone(self.widget.model)
+        self.assertEqual(self.widget.tree_type, self.widget.REGRESSION)
+        self.assertIsNotNone(self.widget.tree_adapter)
```
Comment (kernc) [inline Orange/widgets/visualize/tests/test_owpythagorastree.py]: These kinds of tests seem awfully implementation dependent and, I may be mistaken, totally useless.

Instead, rather check that for sending in a tree model of depth 1, there are at most three SquareGraphicsItems in the view etc.

File: Orange/widgets/visualize/tests/test_owpythagorastree.py [hunk 3]
```diff
@@ -1,12 +1,22 @@
 """Tests for the Pythagorean tree widget and associated classes."""
 import math
 import unittest
+from unittest.mock import Mock
 
+from Orange.classification import TreeLearner
+from Orange.data import Table
+from Orange.regression.tree import TreeRegressionLearner
```
Comment (janezd) [inline Orange/widgets/visualize/tests/test_owpythagorastree.py]: After rebasing to master, you'll need to change this to
`from Orange.regression.tree import TreeLearner as TreeRegressionLearner`

Comment (pavlin-policar) [inline Orange/widgets/visualize/tests/test_owpythagorastree.py]: Thanks for the heads up!

**Comment by pavlin-policar** [comment]
I've removed the bad tests and replaced them with ones that test the UI options. Please take a look again.

**Pull request merged**